### ISSUE TITLE
Implement accessibility (a11y) improvements

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/oik-nivo-slider#16]
Implement accessibility improvements, primarily for the front end.

### Requirements
- Satisfy WCAG 2.1 AA
- See also https://github.com/bobbingwide/oik/issues/79

### Proposed solution

- Update and PHPUnit test with shared library changes for oik, oik-libs and other plugins which deliver shared libraries.
- Update PHPUnit tests to support PHPUnit 10 & 11.
- Test with WAVE and/or axe Dev Tools

**Comment by bobbingwide** [comment]
The main problems with the Nivo slider are:
- Missing alternative text for the images generated in the JavaScript
- Color contrast for the Prev and Next buttons

For https://s.b/oikcom/oik-plugins/oik-nivo-slider-jquery-nivo-slider/ I added a CSS block
```
.theme-default a.nivo-prevNav,
.theme-default a.nivo-nextNav { opacity: 0.5; color: #ffffff; background-color: #000000; border-radius: 50%}
```

For https://s.b/oikcom/oik_shortcodes/nivo/

- the Prev and Next buttons were showing the correct colours when the slider is hovered over
- but there's a Color Contrast issue with the greeny color `#6a6a2a00` due to the `opacity: 0`. 
- this slider is using theme-oik, which doesn't define the colours for `nivo-prevNav` or `nivo-nextNav` so they inherit the link color defined in the theme ( -wp--preset--color--greeny ).
- Also note that the image for the prev and next arrows uses the light green colour

<img width="991" height="336" alt="Image" src="https://github.com/user-attachments/assets/e061772d-1619-400e-aafd-14cea2593e97" />

The fix is similar to that applied in custom CSS for the default theme.
```
.theme-oik a.nivo-prevNav,
.theme-oik a.nivo-nextNav { opacity: 1; color: #000000; background-color: #ffffff; border-radius: 50%}
```

**Comment by bobbingwide** [comment]
After fixing some of the bugs in the jQuery code I still had some problems in 
https://s.b/oikcom/shortcode_example/fix-height-nivo-slider-default-theme/

I did a search in GitHub and found the following issue that seemed relevant

https://github.com/Codeinwp/Nivo-Slider-jQuery/issues/408

**Comment by bobbingwide** [comment]
I applied the changes that had been made to my own copy but I still couldn't get rid of the
Linked image missing alternative text error message.

<img width="1915" height="1021" alt="Image" src="https://github.com/user-attachments/assets/04971ff0-190d-4455-a48e-9cc87e16f457" />

On this URL sometimes I get one message, other times I get two.
The code that's being displayed shows an img tag with the alt attribute set. 
Is this a problem with the WAVE tool being run at the same time as the Nivo jQuery code?

**Comment by bobbingwide** [comment]
**Note:**  the oik nivo slider settings page gets 48 Alerts.
- 46 Redundant links, mostly from the dashboard menu  
- 1  Suspicious link text - due to the fact that I still have the Link CPT enabled
- 1 Layout table 

I can deal with the layout table by changing the code to use grid layout.

**Comment by bobbingwide** [comment]
There were some more Low Color Contrast errors with the oik theme for the slider.
- changed `#359315` to `#286f11`
- changed  `#159374` to `#116f57` 
- I left the border color as it was.
